### PR TITLE
add detailed upgrade steps

### LINKED ISSUE #352: upgrade documentation is way too sparse
I got a operator instance running (v0.2.2) with a production mongoDB cluster.
I tried to figure out what to do to upgrade to latest 0.5.1 in my test environment, but aparently there were several versions needing manual steps and no real documentation for the whole thing.

**What did you expect?**
Documentation describing the different migration steps needed.

**What happened instead?**
Some unclear Notes about mongo resource requiring a new value and "we renamed the resource".

**Operator Information**
 - Operator Version 0.2.2
 - MongoDB Image used 4.2.10

### PR DESCRIPTION
**Pull request by fireba11**
closes #352

### CODE REVIEW

File: docs/install-upgrade.md [hunk 1]
```diff
@@ -155,12 +155,47 @@ To install the MongoDB Community Kubernetes Operator:
...
-   kubectl apply -f config/crd/bases/mongodbcommunity.mongodb.com_mongodbcommunity.yaml
-   ```
+1. Prepare for the upgrade.
+
+   a. Migrate your cr by updating apiVersion and kind to
+      ```
+      apiVersion: mongodbcommunity.mongodb.com/v1
+      kind: MongoDBCommunity
+      ```
+      If you upgrade from pre-0.3.0 you need to also add the field spec.users[n].scramCredentialsSecretName for each resource. This will be used to determine the name of the generated secret which stores MongoDB user credentials. This field must comply with DNS-1123 rules (must consist of lower case alphanumeric characters, '-' or '.', and must start and end with an alphanumeric character).
```
Comment (rodrigovalin) [inline docs/install-upgrade.md:169]: Can you reference https://kubernetes.io/docs/concepts/overview/working-with-objects/names/ and maybe remove the parenthesis comment.

File: docs/install-upgrade.md [hunk 2]
```diff
@@ -155,12 +155,47 @@ To install the MongoDB Community Kubernetes Operator:
...
+      kubectl delete statefulsets.apps mongodb
+      ```
+   c. Delete the old customResourceDefinition. Not strictly needed but no need to keep it around anymore (unless you got more installations of operator in your cluster)
+      ```
+      kubectl delete crd mongodb.mongodb.com
+      ```
+3. In with the new
+
+   Follow the normal installation procedure above.
+4. Start up your cluster again
```
Comment (rodrigovalin) [inline docs/install-upgrade.md:188]: Maybe specify that you are referring to the MongoDB cluster or Replica Set. `cluster` can be confused with "Kubernetes cluster".

File: docs/install-upgrade.md [hunk 3]
```diff
@@ -155,12 +155,47 @@ To install the MongoDB Community Kubernetes Operator:
...
+   a. Re-create your cr using the new Version from Step 1.a
+   b. Patch your statefulset to have it update the permissions
```
Comment (rodrigovalin) [inline docs/install-upgrade.md:190]: Is this something specific you need for your environment? The Operator should create the StatefulSet with the right SecurityContext at this stage, and I'm not sure why you would need to patch it manually.

**Review by rodrigovalin — COMMENTED** [review]
Hello and thanks for your contribution!

This is looking really good; I have left some comments and I would really appreciate if you could address them.

Thanks again

File: docs/install-upgrade.md (hunk 3 above)
Comment (fireba11) [reply docs/install-upgrade.md:190]: The patch is from release notes of v0.6.0 to fix filesystem rights on the PV.
It is actually needed for mongod to be able to start with the old data.

File: docs/install-upgrade.md (hunk 1 above)
Comment (fireba11) [reply docs/install-upgrade.md]: Replaced the text in parenthesis with the reference link.

File: docs/install-upgrade.md (hunk 2 above)
Comment (fireba11) [reply docs/install-upgrade.md]: changed

**Review by rodrigovalin — APPROVED** [review]
Thanks for your contribution @fireba11

**Comment by rodrigovalin** [comment]
@fireba11 we only need for your commits to be signed before being able to merge the patch.

**Comment by fireba11** [comment]
@rodrigovalin i realized. first time that was required, i just set up a key to do that ... and rebase the commits to sign them (at least that's the way i found to retroactively do that). hope i did everything correct :-)

**Comment by bznein** [comment]
Hi @fireba11 

the signing looks correct now! Will merge tomorrow after all the required checks pass :)

Thanks again for your contribution!

**Pull request merged**